pkgs/development/python-modules/cyipopt/default.nix

```diff
@@ -0,0 +1,43 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  cython,
+  ipopt,
+  numpy,
+  pkg-config,
+  setuptools,
+}:
+
+buildPythonPackage rec {
+  pname = "cyipopt";
+  version = "1.5.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "mechmotum";
+    repo = "cyipopt";
+    rev = "v${version}";
+    hash = "sha256-ddiSCVzywlCeeVbRJg2wxKIlAVlZw9Js95IbEDqhh5Q=";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ ipopt ];
+
+  build-system = [
+    cython
+    numpy
+    setuptools
+  ];
+
+  pythonImportsCheck = [ "cyipopt" ];
+
```

Comment:
Could you enable tests with `pytestCheckHook`?

Reply:
Ok, that is here: yes: `python3.12-cyipopt> ================== 30 passed, 63 skipped, 1 warning in 0.31s ===================`

Reply:
Oh right, they run automatically.
I think that it is better to explicitly add
```nix
nativeCheckInputs = [
  pytestCheckHook
];
```

Reply:
Sure, it makes more sense, done, thanks.
And I still have this `python3.12-cyipopt> ================== 30 passed, 63 skipped, 1 warning in 0.31s ===================`